libethcore/Transaction.cpp

```diff
@@ -64,10 +64,15 @@ TransactionBase::TransactionBase(bytesConstRef _rlpData, CheckTransaction _check
 			BOOST_THROW_EXCEPTION(InvalidTransactionFormat() << errinfo_comment("transaction data RLP must be an array"));
 
 		m_data = rlp[field = 5].toBytes();
-		byte v = rlp[field = 6].toInt<byte>() - 27;
+
+		byte v = rlp[field = 6].toInt<byte>();
 		h256 r = rlp[field = 7].toInt<u256>();
 		h256 s = rlp[field = 8].toInt<u256>();
 
+		m_chainId = std::floor((float)(v - 35)/2);
```

Comment:
Please don't use float, it might create some precision problems. What about
`(int(unsigned(v)) - 35) / 2`?